packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -368,53 +370,28 @@ export abstract class FluidDataStoreContext extends EventEmitter implements
     }
 
     /**
-     * Notifies the object to take snapshot of a store.
-     * @deprecated in 0.22 summarizerNode
+     * Returns a summary at the current sequence number.
+     * @param fullTree - true to bypass optimizations and force a full summary tree
+     * @param trackState - This tells whether we should track state from this summary.
      */
-    public async snapshot(fullTree: boolean = false): Promise<ITree> {
-        if (!fullTree) {
-            const id = await this.summaryTracker.getId();
-            if (id !== undefined) {
-                return { id, entries: [] };
-            }
-        }
-
-        await this.realize();
-
-        const { pkg, isRootDataStore } = await this.getInitialSnapshotDetails();
-
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        const entries = await this.channel!.snapshotInternal(fullTree);
-
-        const attributesBlob = createAttributesBlob(pkg, isRootDataStore);
-        entries.push(attributesBlob);
-
-        return { entries, id: null };
-    }
-
-    public async summarize(fullTree = false): Promise<ISummarizeResult> {
-        return this.summarizerNode.summarize(fullTree);
+    public async summarize(fullTree: boolean = false, trackState: boolean = true): Promise<ISummarizeResult> {
```

Comment:
I wonder if environment (fullTree, trackState) should be passed as an object, and always go through summarizerNode that does the pivot on tracking vs. not

Reply:
I debated whether to move this to summarizerNode or not and discussed it with Arin. Both approaches have their pros and cons but Arin slightly preferred that the summarizerNode always track state. So, I kept it this way for now.
I think this can be easily changed since it doesn't have backward compat concerns and we can do it later if we want.